XiEditor/EditView.swift

```diff
@@ -436,7 +444,7 @@ class EditView: NSView, NSTextInputClient {
     /// timer callback to toggle the blink state
     @objc func _blinkInsertionPoint() {
         _cursorStateOn = !_cursorStateOn
-        needsDisplay = true
+        partialInvalidate(invalid: dataSource.lines.cursorInval)
```

Comment:
A minor improvement possible here: `EditView` could just have a `invalidationSetForCurrentCursor` property that is set when the update arrives, which would save recomputing on each blink.

Not a blocker, more an observation.

Reply:
I agree this is a good optimization, but I expect the impact to be tiny.

Reply:
agreed, just an observation. :)